rllib/algorithms/ppo/ppo_learner.py

```diff
@@ -39,6 +47,78 @@ def build(self) -> None:
             )
         )
 
+    @override(Learner)
+    def _preprocess_train_data(
+        self,
+        *,
+        batch,
+        episodes,
+    ):
+        batch = batch or {}
+        if not episodes:
+            return batch, episodes
+
+        # Make all episodes one ts longer in order to just have a single batch
```

Comment:
New way to do GAE:
* elongate all episodes by one artificial ts.
* perform vf-predictions AND bootstrap value predictions in one single batch (b/c we have the extra timestep!)
* * use the learner connector to make sure this forward pass is done using the correct (custom?) batch format.
* remove extra timesteps from episodes (and computed advantages)